### PR TITLE
DS-3321 Allow AIP Import of Item with Embargo Date in the Past

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3321

### CODE REVIEW

**Comment by terrywbrady** [comment]
[13:34:20]	‹tdonohue›  terry-b3: PR 1537 looks pretty small...just catches the error to ignore it. Seems reasonable
[13:35:29]	‹terry-b3›  The only risk I see is if someone improperly formatted a lift date ... it could go through without the user noticing the error
[13:40:03]	‹tdonohue›  so it's really just 1537. I agree, terry-b3, it would also ignore improperly formatted lift dates. Not sure if there's any other way to achieve this bug fix though (off top of my head)
[13:40:44]	‹terry-b3›  The other approach would be to throw a different exception for the lift date in the past. Or move that check.
[13:40:57]	‹terry-b3›  I am not sure how valuable the effort would be
[13:41:14]	‹mhwood›  Yes, I was just thinking that EmbargoDatePassed could be a separate exception.
[13:42:34]	‹mhwood›  Or setEmbargo could take a boolean to say whether we care about passed dates. This *is* restoring from backup, after all, and the rules are different.
[13:43:11]	‹mhwood›  (Or just have a new method restoreEmbargo.)
[13:43:50]	‹terry-b3›  I like the 3rd option best.
[13:44:07]	‹terry-b3›  (restoreEmbargo)
[13:44:16]	‹terry-b3›  if you all concur, I will make the change
[13:44:20]	‹tdonohue›  So, I notice that setEmbargo *does* have some comments about *restoring* items from Backup... I wonder if it's actually got a bug here
[13:44:25]	‹tdonohue›  https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java#L86
[13:44:26]	‹kompewter›  [ DSpace/EmbargoServiceImpl.java at master · DSpace/DSpace · GitHub ] - https://github.com/DSpace/DSpace/blob/master/dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java#L86
[13:45:02]	‹tdonohue›  I'd lean towards looking closer at setEmbargo code (seems to imply it should support restoration). But, if that's not possible, a restoreEmbargo() method seems fine to me
[13:45:30]	‹mhwood›  I was just guessing that there are calls where we *do* care about passed dates.
[13:46:14]	‹terry-b3›  I will look at it more closely and revise the PR. Thanks for the reviews on these
[13:46:18]	‹tdonohue›  so, it seems like we have a good direction / next step for 1537 then.

**Comment by terrywbrady** [comment]
Note EmbargoServiceImpl.setEmbargo() calls EmbargoSetter.setEmbargo() which calls EmbargoServiceImpl.getEmbargoTermsAsDate().  

In order to catch a custom exception in EmbargoSetter.setEmbargo(), all implementing instances would need to check for the new exception type.

When calling EmbargoSetter.setEmbargo(), the recovery context is no longer known.

**Comment by hardyoyo** [comment]
+1 by inspection

File: dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java [hunk 1]
```diff
@@ -142,9 +142,9 @@ public DCDate getEmbargoTermsAsDate(Context context, Item item)
         // sanity check: do not allow an embargo lift date in the past.
         if (liftDate.before(new Date()))
         {
-            throw new IllegalArgumentException(
-                    "Embargo lift date must be in the future, but this is in the past: "
-                            + result.toString());
+            log.info("Embargo lift date must be in the future, but this is in the past");
+            //DS-3321, existing code assumes a null lift date when in the past
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java]: Not sure I understand the context behind this change and the comment.  In 5.x, `EmbargoManager.getEmbargoTermsAsDate()` also threw this `IllegalArgumentException` if a date was in the past, see: https://github.com/DSpace/DSpace/blob/dspace-5_x/dspace-api/src/main/java/org/dspace/embargo/EmbargoManager.java#L154

This makes me slightly suspicious that something else is going on here.  The code logic in *both* `getEmbargoTermsAsDate()` AND `setEmbargo()` appears nearly identical between 5.x and master.  

It is possible there was some "hidden" bug in this logic that we never encountered in 5.x (but we are encountering now in the refactored API).  But, I'd like us to better understand what is going on  here, so we can say with certainty that this exception should *not* be thrown in this scenario.

Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/embargo/EmbargoServiceImpl.java]: A few clarifications to my questions above..

To be fair, I'm not against the changes suggested in the PR.  It's just that this is throwing up potential red flags in my mind... and I want to be certain we have our reasons right...currently it seems to be a "this worked, so maybe it's right" sorta PR.  So, I'm _not_ a -1.  But, I am currently a 0 vote, and I'd need further convincing to move to a +1 vote.

**Comment by terrywbrady** [comment]
Based on more careful testing on DSpace 5, I am able to reproduce the error in DSpace 5.  I will update the ticket.  We may decide that deserves reconsideration.

**Pull request closed without merging**